config/config.go

```diff
@@ -45,19 +79,26 @@ type Config struct {
 	K2 uint32 // K2 is the number of labels below the required difficulty required for a proof.
 	K3 uint32 // K3 is the size of the subset of proof indices that is validated.
 
+	// FIXME: remove support for the old scrypt-based PoW
 	K2PowDifficulty uint64
+	PowDifficulty   [32]byte
 }
 
 func DefaultConfig() Config {
-	return Config{
+	cfg := Config{
 		LabelsPerUnit:   512, // 8kB units
 		MaxNumUnits:     defaultMaxNumUnits,
 		MinNumUnits:     defaultMinNumUnits,
-		K1:              273,
-		K2:              300,
-		K3:              100,
+		K1:              26,
+		K2:              37,
+		K3:              37,
 		K2PowDifficulty: 0x0FFFFFFF_FFFFFFFF,
 	}
+	for i := range cfg.PowDifficulty {
+		cfg.PowDifficulty[i] = 0xFF
+	}
+
+	return cfg
```

Comment:
With `PowDifficulty` as a slice this can be written more concisely as:

```golang
	return Config{
		LabelsPerUnit:   512, // 8kB units
		MaxNumUnits:     defaultMaxNumUnits,
		MinNumUnits:     defaultMinNumUnits,
		K1:              26,
		K2:              37,
		K3:              37,
		K2PowDifficulty: 0x0FFFFFFF_FFFFFFFF,
		PowDifficulty:   bytes.Repeat(0xFF, 32),
	}
}
```

The check for the length then has to be done in `GenerateProof` and `VerifyProof` (where we are already checking for other lengths as well).

It would also allow to pass the difficulty without the need to copy it first (see other comment).